### ISSUE TITLE
Add an Item type for character classes

### ISSUE THREAD

**Issue by wyrmisis** [vttred/ose#283]
### New Feature or Changed Behaviour

I think we should add a new Item type for character classes. 

When this Item is added to an Actor, an Actor of `character` type will no longer manage its own level, XP to next level, XP bonus from prime requisites, save values, spell slots, hit dice, attack bonus, or whether or not the Actor is a spellcaster. This will instead be informed by properties on the Character Class Item.

When added to a non-`character` Actor, this Item will be rejected, and not added to the Actor's list of Items.

The Character Class item will contain the following:
* A tab containing a grid of levels, XP values, attack bonuses, save values, and (optionally) spell slots.
* A tab containing config options for class requirements, prime requisites (and prime requisite XP bonuses), and languages
* A tab containing Items of type `ability` to be added when the class is dropped in and removed when the class is removed, as well as RollTables to be rolled when the Character Class Item is dropped in (for folks who roll for gear or starting spell)

Once dragged in, the Character Class item will appear similarly to how we display the Treasure Table on a Monster sheet, and can be removed by right clicking it.

### Why?

This addition would help GMs bring in new classes more easily and automate advancement for players. I think it would help tighten the character creation loop and reduce potential data entry errors for players and GMs.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bakbakbakbakbak** [comment]
Yes! This would also increase usability when transitioning to the UFT, as it would help people only interested in game design to work on just that, and not having to go into system development!
